pkg/reconciler/revision/revision.go

```diff
@@ -81,20 +82,40 @@ func (c *Reconciler) reconcileDigest(ctx context.Context, rev *v1.Revision) erro
 	if rev.Status.ImageDigests == nil {
 		rev.Status.ImageDigests = make(map[string]string, len(rev.Spec.Containers))
 	}
-	for _, container := range rev.Spec.Containers {
-		digest, err := c.resolver.Resolve(container.Image,
-			opt, cfgs.Deployment.RegistriesSkippingTagResolving)
-		if err != nil {
-			err = fmt.Errorf("failed to resolve image to digest: %w", err)
-			rev.Status.MarkContainerHealthyFalse(v1.ReasonContainerMissing,
-				v1.RevisionContainerMissingMessage(
-					container.Image, err.Error()))
-			return err
+
+	var digestGrp errgroup.Group
+	type digestData struct {
+		digestValue   string
+		containerName string
+		containerPort int
+	}
+
+	digests := make(chan *digestData, len(rev.Spec.Containers))
```

Comment:
```suggestion
	digests := make(chan digestData, len(rev.Spec.Containers))
```

Don't pass pointers here, our data is relatively small anyway.